File: BodyMICalc/app/src/main/java/com/s16665/bodymicalc/MainActivity.java
package com.s16665.bodymicalc;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;

import android.os.Bundle;
import android.view.MenuItem;

import com.google.android.material.bottomnavigation.BottomNavigationView;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        com.google.android.material.bottomnavigation.BottomNavigationView bottomNavigationView = findViewById(R.id.bottomNav);
        bottomNavigationView.getMenu().getItem(0).setChecked(true);
        bottomNavigationView.setOnNavigationItemSelectedListener(navigationItemSelectedListener);
        getSupportFragmentManager().beginTransaction().replace(R.id.fragmentLayout, new StartFragment()).commit();
    }

    private BottomNavigationView.OnNavigationItemSelectedListener navigationItemSelectedListener = new BottomNavigationView.OnNavigationItemSelectedListener() {
        @Override
        public boolean onNavigationItemSelected(@NonNull MenuItem item) {
            Fragment selectedFragment = new StartFragment();
            switch (item.getItemId()) {
                case R.id.start:
                    selectedFragment = new StartFragment();
                    break;
                case R.id.bmi:
                    selectedFragment = new BmiFragment();
                    break;
                case R.id.food:
                    selectedFragment = new FoodFragment();
                    break;
                case R.id.chart:
                    selectedFragment = new ChartFragment();
                    break;
                case R.id.quiz:
                    selectedFragment = new QuizFragment();
                    break;
            }
            getSupportFragmentManager().beginTransaction().replace(R.id.fragmentLayout, selectedFragment).commit();
            return true;
        }
    };
}

File: BodyMICalc/app/src/main/java/com/s16665/bodymicalc/MainActivity.kt
package com.s16665.bodymicalc

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import com.google.android.material.bottomnavigation.BottomNavigationView

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val bottomNavigationView = findViewById<BottomNavigationView?>(R.id.bottomNav)
        bottomNavigationView?.getMenu()?.getItem(0)?.isChecked = true
        bottomNavigationView?.setOnNavigationItemSelectedListener(navigationItemSelectedListener)
        supportFragmentManager.beginTransaction().replace(R.id.fragmentLayout, StartFragment()).commit()
    }

    private val navigationItemSelectedListener: BottomNavigationView.OnNavigationItemSelectedListener? = BottomNavigationView.OnNavigationItemSelectedListener { item ->
        var selectedFragment: Fragment = StartFragment()
        when (item.itemId) {
            R.id.start -> selectedFragment = StartFragment()
            R.id.bmi -> selectedFragment = BmiFragment()
            R.id.food -> selectedFragment = FoodFragment()
            R.id.chart -> selectedFragment = ChartFragment()
            R.id.quiz -> selectedFragment = QuizFragment()
        }
        supportFragmentManager.beginTransaction().replace(R.id.fragmentLayout, selectedFragment).commit()
        true
    }
}